docs/04-api-reference/front-commerce-remix/index.mdx

```diff
@@ -91,33 +124,75 @@ Instead of the standard remix `SerializeObject` type:
 SerializeObject<UndefinedToOptional<{ foo: "bar"; date: string }>>;
 ```
 
-## Front-Commerce Context
+### `useApiFetcher`
 
-To ensure the Front-Commerce Context is available in your loaders and actions,
-you need to add the `createFrontCommerceContext` to your server via the
-[`getLoadContext`](https://remix.run/docs/en/2.0.1/route/loader#context), to do
-this, update your
-[`server.ts`](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/b99f0475ce494591a54fdc2b904e956a083e9b51/skeleton/server.ts)
+The `useApiFetcher` is an optimised fetcher which allows to deduplicate client
+side calls the the same `/api` endpoint. You can replace a load of `useFetcher`
+calls with a single `useApiFetcher` call with a simplified API for loading
+loader data.
 
-```ts
-// add-start
-import { createFrontCommerceContext } from "@front-commerce/remix";
-import { createRemixRequest } from "@remix-run/express/dist/server";
-// add-end
+Here is an example of how to change from `useFetcher` to `useApiFetcher`:
 
-app.all(
-  "*",
-  createRequestHandler({
-    build: require(BUILD_DIR),
-    mode: process.env.NODE_ENV,
-    // add-start
-    getLoadContext(request, response) {
-      const remixRequest = createRemixRequest(request, response);
-      return {
-        frontCommerce: createFrontCommerceContext(remixRequest),
-      };
-    },
-    // add-end
-  })
-);
+```tsx title="/app/routes/my-route.tsx"
+import type { loader } from "./api.cart";
+// remove-next-line
+import { useFetcher } from "@front-commerce/remix/react";
+// add-next-line
+import { useApiFetcher } from "@front-commerce/remix/react";
+
+export default function MyRoute() {
+  // remove-next-line
+  const fetcher = useFetcher<typeof loader>();
+  // add-next-line
+  const { data, loading } = useApiFetcher<typeof loader>("/api/cart");
+
+  // remove-start
+  useEffect(() => {
+    if (!fetcher.data && fetcher.state !== "loading") {
+      fetcher.load("/api/cart");
+    }
+  }, [fetcher]);
+  // remove-end
+
+  // remove-next-line
+  if (fetcher.state === "loading") {
+    // add-next-line
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  // remove-next-line
+  const data = fetcher.data
+
+  return (
+    <div>
+      Cart Id: {data.cart.id}
+      Cart Qty: {data.cart.items_qty}
+    </div>
+  );
+}
 ```
+
+Here is an example of 5 concurrent calls to the same `/api/cart` endpoint:
+
+<BrowserWindow noPadding>
+
+<video
+  width="100%"
+  height="400"
+  controls
+  className="video video-contain"
+  style={{ marginBottom: "-8px" }}
+>
+  <source
+    src={require("./assets/useApiFetcher.mp4").default}
+    type="video/mp4"
+  />
+  Your browser does not support the video tag.
+</video>
+
+</BrowserWindow>
+
+As you can see with the `useFetcher` it will call the api for each component,
+with the `useApiFetcher` it will only called the api once and share the data
+between the components.
```

Comment:
```suggestion
As you can see with the `useFetcher` it will call the API for each component,
with the `useApiFetcher` it will only called the API once and share the data
between the components.
```

Reply:
Also, don't you think it'd be worth to add a line about revalidation? (it will also prevent multiple calls in this case)